week_2/pet.rb

```diff
@@ -110,6 +127,13 @@ def self.cost_to_feed(pets, days)
   #
   # Note - Order is not important
   def self.group_by_animal_type(pets)
-    raise NotImplementedError # TODO
+    res = {}
+    pets.each { |pet|
+      res[pet.animal_type_id] = []
+    }
+    pets.group_by { |pet| 
+      res[pet.animal_type_id].push(pet.name)
+    }
```

Comment:
Wrong use of [`group_by`](http://jelera.github.io/howto-work-with-ruby-group-by), group_by is used  for grouping items from an enumerable and return a hash. Since you are just iterating `each` itself can be used here.

You can also use a single `each` block to so this.
```ruby
pets.each do |pet|
  if res[pet.animal_type_id]
      # ....
  else
      res[pet.animal_type_id] = [pet.name]
   end
end  
```